### PR TITLE
Update/account balances

### PR DESCRIPTION
**Pull request by dutterbutter**
**Changes made in this pull request:**

If I can get someone to review and double check that the account balances are being updated with the correct values that would be awesome. Still have to write unit tests for these functions (genesis functions and internal txs, and account balances). To create internal txs,` shyftDb > web3> transfer_through_master > then run truffle test `while geth is running. Regular txs run `./shyft-geth.sh --js sendTransactions`

Also, the accountBalance updating functions in shyft_db_utils could use some refactoring. Currently, we have 3 distinct accountBalance updating fns, 1 for regular txs, 1 for contract creation txs, 1 for internal txs. I feel we can consolidate into one general purpose fn. 

- Refactored write functions in genesis.go (where we write block 0 and gen.Alloc accounts)
- Added a switch case so our ShyftGenesis functions are only called once (geth calls SetUpGenesisBlock twice for chaindata, and lightchaindata)
- Storing all hashes in lowercase for equality reasons
- Added internalTx write db utility function 
- Added blockExists read db utility function
- Accounts are being updated based on internal Txs value transfers

### CODE REVIEW

**Comment by GregTheGreek** [comment]
I just spent a good time analyzing the smart contracts and the txs and it seems to be incorrect for some reason posting a photo shortly... Haven't gotten to the accounts yet.

**Comment by GregTheGreek** [comment]
As you can see there seems to be 1 additional accounts that are undocumented...
The eth seems to go to the right person but for whatever reason the contract addr for Transfer2 is different then that listed fromthe test

screen shot 2018-07-29 at 12 40 07

**Comment by dutterbutter** [comment]
All addresses are being documented but still need a review on account balances.

**Comment by GregTheGreek** [comment]
I will check the account balances after our scrum call.

**Comment by GregTheGreek** [comment]
I'm seeing the correct balances, on my end. @strangepowers @dbkbali anything from you guys?

**Comment by dutterbutter** [comment]
I am gunna push one change that will need a review, basically in the switch case for updating balances, if a to address does not exist we create that account with that balance but the from address balance is never updated. I am gunna push that change momentarily.

**Comment by dutterbutter** [comment]
Account balances are being updated seemingly correctly, we are now calling SWriteAccountBalances which updates the TO_ADDRESS and FROM_ADDRESS balances through the internalTx Tracer. However, the unit tests for accounts and account balances are incomplete as doing it this way introduces some complexities to setting up a proper testing environment.

**Comment by GregTheGreek** [comment]
@dutterbutter I still think we could get the go test to execute the internal transaction js file and just compare that after the file executed that the internal balances update by the exact amount...

**Comment by dutterbutter** [comment]
I can try it but I believe you need Geth to be running while the js file is executed.

**Comment by GregTheGreek** [comment]
This is true. We can look into making a script do this or something later.
Let's merge this and move forward.

**Comment by dutterbutter** [comment]
@GregTheGreek updated changes.

File: eth/shyft_tracer.go [hunk 1]
```diff
@@ -23,26 +24,27 @@ func InitTracerEnv() {
...
-		Tracer: &jsTracer,  // needs to be non-nil
-		Timeout: nil,
-		Reexec: nil,
+		Tracer:    &jsTracer, // needs to be non-nil
+		Timeout:   nil,
+		Reexec:    nil,
 	}
 	TracerConfig = config
 	fullNode, _ := SNew(Global_config)
+	fmt.Println("FULL NODE", fullNode)
```
Comment (GregTheGreek) [inline eth/shyft_tracer.go]: do we need this?

**Pull request merged**